### ISSUE TITLE
Make a demo SBT subproject

### ISSUE THREAD

**Issue by mslinn** [mslinn/web3j-scala#2]
Currently the demo runs in `Test` scope; it should instead run in `Runtime` scope, as part of a new SBT project called `demo` that is dependent on the `root` project. The `root` project should publish the `web3j-scala` library. 

`build.sbt` needs a new task that publishes a fat jar for the demo (the `web3j-scala-demo` project), and it should also publish in parallel with the library in the root project. Both projects should have synchronized version numbers.
